openedx/core/djangoapps/user_authn/views/register.py

```diff
@@ -216,6 +216,11 @@ def create_account_with_params(request, params):
     create_user_link_with_edly_sub_organization(request, user)
     user = create_learner_link_with_permission_groups(user)
 
+    # Sites using multiple languages need to record the language used during registration.
+    # If not, compose_and_send_activation_email will be sent in site's default language only.
+    create_or_set_user_attribute_created_on_site(user, request.site)
+    preferences_api.set_user_preference(user, LANGUAGE_KEY, get_language())
+
```

Comment:
didn't we make this change in previous PR? maybe branch needs to be rebased.

Reply:
@taimoor-ahmed-1 fixed. The PR wasn't synced with develop-juniper after the rebased